Automattic/wp-calypso
# Social: Hide the basic plan from the jetpack cloud page.

## Proposed Changes

* Hide the product modal on the pricing page from showing the basic plan 

<img width="529" alt="Screenshot 2024-04-10 at 6 44 18 PM" src="https://github.com/Automattic/wp-calypso/assets/6594561/f9c3dc64-46b4-4876-b2fc-d146dd67add9">

<img width="948" alt="Screenshot 2024-04-10 at 6 44 22 PM" src="https://github.com/Automattic/wp-calypso/assets/6594561/6ba850b5-83bc-47fe-bf19-d95760e6ff75">

## Testing Instructions



* Go to http://jetpack.cloud.localhost:3000/pricing/ after booting up your local Jetpack cloud. Alternatively, you could use the live link. 
* Click on the more about Social Button and see the modal that loads. You should only see the advanced plan it. 
* Click on the more about GET Button and see that you're redirected to the checkout page. 
* Buy the Jetpack Social Basic monthly plan on your site - you can do this on Jetpack Cloud since you can't buy the basic plan via this branch since we are hiding it. Come to the pricing page http://jetpack.cloud.localhost:3000/pricing/<site-id> and click on the manage subscription button next to Social and ensure you're redirected to the purchase screen. 

* Buy the Jetpack Social advanced subscription on a different site and come back to he pricing page http://jetpack.cloud.localhost:3000/pricing/<site-id> and click on the manage subscription button next to Social and ensure you're redirected to the purchase screen. 


## Pre-merge Checklist
